app/src/main/java/com/tomaszpolanski/androidsandbox/ViewModelBase.kt

```diff
@@ -0,0 +1,25 @@
+package com.tomaszpolanski.androidsandbox
+
+import android.arch.lifecycle.ViewModel
+import android.os.Parcelable
+import io.reactivex.Flowable
+import io.reactivex.Single
+import io.reactivex.processors.PublishProcessor
+
+abstract class ViewModelBase<S : Parcelable, in E>(
```

Comment:
I generally like this API - I tried to do something similar with my take on MVI, but I got distracted by implementation details.

Reply:
Nice to hear that :)